frontend/simple/main.js

```diff
@@ -28,11 +31,19 @@ var loginGuard = {
   guard: store => !store.state.loggedIn,
   redirect: (to, from) => ({ path: '/signup', query: { next: to.path } })
 }
+var signupGuard = {
+  guard: store => !!store.state.loggedIn,
+  redirect: (to, from) => ({ path: '/' })
+}
 // Check if user has a group to invite users to
 var inviteGuard = {
   guard: store => !store.state.currentGroupId,
   redirect: (to, from) => ({ path: '/new-group' })
 }
+var joinGuard = {
+  guard: (store, to, from) => from.name !== Mailbox.name,
+  redirect: (to, from) => ({ path: '/mailbox' })
+}
```

Comment:
There are so many guards cluttering up the file now. These look like they should belong in the route definition itself below, not here.